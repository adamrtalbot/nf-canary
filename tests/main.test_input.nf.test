nextflow_process {

    name "Test Process TEST_INPUT"
    script "main.nf"
    process "TEST_INPUT"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.of("alpha").collectFile(name: 'sample.txt', newLine: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
