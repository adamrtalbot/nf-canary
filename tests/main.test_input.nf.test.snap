{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "alpha\n"
                ]
            }
        ],
        "timestamp": "2023-07-20T10:49:51+0000"
    }
}